mage_integrations/mage_integrations/sources/google_ads/__init__.py

```diff
@@ -20,28 +17,16 @@ def discover(self, streams: List[str] = None) -> Catalog:
 
         catalog = do_discover(resource_schema)
 
-        catalog_entries = []
-
-        for stream in catalog['streams']:
-            stream_id = stream['tap_stream_id']
-            if not streams or stream_id in streams:
-                schema = catalog_singer.Schema.from_dict(stream['schema'])
-                catalog_entries.append(self.build_catalog_entry(
-                    stream_id,
-                    schema,
-                    **ignore_keys(stream, ['schema']),
-                ))
-
-        return Catalog(catalog_entries)
+        return Catalog(catalog['streams'])
```

Comment:
[catalog_entries](https://github.com/mage-ai/mage-ai/pull/4099/files#diff-3ab672cac903517d4f4d3b0b3b8c9c00bfd3782d42cebdfedd0d3733251b5af1L29) generation does not append stream metadata that's generated in [do_discover_streams](https://github.com/mage-ai/mage-ai/blob/2c3b97670fe3f60bc56478f45b6ad52deedf9f84/mage_integrations/mage_integrations/sources/google_ads/tap_google_ads/discover.py#L243). Also, all streams are already discovered by the `do_discover` tap method.
I belive that the `streams` parameter is mainly used due to the possible high discover time of some specific taps. Since the discover will always be executed on all streams, instead of appending the metadata generation to this catalog_entries, i deemed ok to just remove the whole double catalog situation and just use the tap-generated catalog